lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -443,146 +517,243 @@ def create_autoscaling_group(connection, module):
     health_check_type = module.params.get('health_check_type')
     default_cooldown = module.params.get('default_cooldown')
     wait_for_instances = module.params.get('wait_for_instances')
-    as_groups = connection.get_all_groups(names=[group_name])
+    as_groups = connection.describe_auto_scaling_groups(AutoScalingGroupNames=[group_name])
     wait_timeout = module.params.get('wait_timeout')
     termination_policies = module.params.get('termination_policies')
     notification_topic = module.params.get('notification_topic')
     notification_types = module.params.get('notification_types')
 
     if not vpc_zone_identifier and not availability_zones:
-        region, ec2_url, aws_connect_params = get_aws_connection_info(module)
+        region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
         try:
-            ec2_connection = connect_to_aws(boto.ec2, region, **aws_connect_params)
-        except (boto.exception.NoAuthHandlerFound, AnsibleAWSError) as e:
+            ec2_connection = boto3_conn(module,
+                                        conn_type='client',
+                                        resource='ec2',
+                                        region=region,
+                                        endpoint=ec2_url,
+                                        **aws_connect_params)
+        except (botocore.exceptions.NoCredentialsError, AnsibleAWSError) as e:
             module.fail_json(msg=str(e))
     elif vpc_zone_identifier:
         vpc_zone_identifier = ','.join(vpc_zone_identifier)
 
     asg_tags = []
     for tag in set_tags:
-        for k,v in tag.items():
+        for k,v in iteritems(tag):
```

Comment:
In Ansible we typically just use `.items()` because as a loop iterator there's no (usage) difference between the Py2 list and Py3 iterable behavior, and if your dictionary is less than 1000 items or so, the iterable behaviour is actually *slower* in Python 2 (trading time for space, naturally). Can you please leave it as `items` instead?